frontend/src/components/page_content/profile/goals.js

```diff
@@ -49,46 +38,37 @@ class Goals extends React.Component {
     return Style.it(
       `${styles}`,
       <div className={this.getClasses()}>
-        <ToastSelect
-          options={goalOptions}
-          name="description"
-          label="Goal 1"
-          list="goals"
-          placeholder="Select your goal"
-          defaultValue={this.props.goals[0].description}
-          id="goals"
-          onChange={this.setGoalListValue}
-        />
-
-        <div className="row">
-          <div className="column">
-            <ToastInput
-              type="number"
-              label="Amount"
-              name="amount"
-              placeholder="1,000"
-              min={0.0}
-              step={0.01}
-              iconName="dollarsign"
-              defaultValue={this.props.goals[0].amount}
-              onChange={this.setGoalListValue}
-            />
-          </div>
-          <div className="column">
-            <ToastInput
-              type="date"
-              label="Goal End Date"
-              defaultValue={this.props.goals[0].endDate}
-              name="endDate"
-              onChange={this.setGoalListValue}
-            />
-          </div>
-        </div>
-
-        {/* TODO: Add inputs on button click */}
-        <Center>
-          <ToastButton tertiary label="Add Goal" />
-        </Center>
+        <ToastDuplicateInputButton
+          label="Add Goal"
+          fields={{ goal: "", amount: "", endDate: "" }}
+          value={this.props.goals}
+          onChange={this.props.setGoalListValue}
+          onDuplicate={this.props.addGoal}
+        >
+          <ToastSelect
+            options={goalOptions}
+            name="description"
+            label="Goal 1"
+            list="goals"
+            placeholder="Type in your goal"
+            id="goals"
+            onChange={this.setGoalListValue}
```

Comment:
i dont think you need this onChange because ToastDuplicateInputButton might handle it?